### PR TITLE
Implement react redux for frontend

### PR DESCRIPTION
**Pull request by gilberthappi**
**Learning objectives**
Implement a connection between a Ruby on Rails back-end and React front-end.
Understand pros and cons of different approaches of connecting Ruby on Rails back-end with React front-end.

**Project requirements**
Create a new React app called 'hello-react-front-end'
Initialize your project with Git.
Create the App component with react-router.
Create the Greeting component that will display a greeting. Set it up as a route in your App component.
Create a store, an action and a reducer that will connect to you API endpont to get the random greeting.
Display the random greeting in your Greeting component.

[BACKEND](https://github.com/gilberthappi/hello-rails-back-end/pull/1)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,174 @@
...
+
+### Key Features <a name="key-features"></a>
+- Making use of Ruby on Rails
+- Making use of PostgreSQL
+
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+<!-- LIVE DEMO -->
+
```
Comment (topeogunleye) [inline README.md:63]: @gilberthappi, Hi I am your code reviewer, You have one thing left, please kindly add the link to your back end also in your README for the frontend:

File: src/App.jsx [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        </p>
+      </div>
+      <p className="read-the-docs">
+        Click on the Vite and React logos to learn more
+      </p>
+    </div>
+  )
+}
+
+export default App
```
Comment (topeogunleye) [inline src/App.jsx]: ---

Hi, this is a good start for your project, but there are some things that you need to improve to meet the requirements. Here are some suggestions:

- You need to import and use `react-router` to create routes for your App component. You can follow this [guide](https://reactrouter.com/web/guides/quick-start) to learn how to do that.
- You need to create a Greeting component that displays a personalized message to the user based on their name or username. You can use props or state to pass the user's name or username to the component.
- You need to add a route that leads to the Greeting component in your App component. For example, you can use `/greeting` as the path for the route. You also need to add a link or a button that navigates to the Greeting component from the App component.
- You need to set up a store, an action, and a reducer that uses the url to the endpoint from the Rails API. You can use `redux` or `react-redux` to manage the state of your application. You can follow this [tutorial](https://redux.js.org/tutorials/essentials/part-1-overview-concepts) to learn how to use redux with react. You can use `axios` or `fetch` to make requests to the Rails API. You can follow this [article](https://www.digitalocean.com/community/tutorials/how-to-connect-to-an-api-with-javascript) to learn how to use axios or fetch with javascript.

I hope this helps you improve your code and meet the project requirements. Good luck! 😊

File: README.md (hunk 1 above)
Comment (gilberthappi) [inline README.md:63]: okey thanks

**Review by topeogunleye — CHANGES_REQUESTED** [review]
Hi @gilberthappi, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

Try-Again

> ## (Highlights) Good Points: 💪 
> * Set Up React Application With VITE :heavy_check_mark:

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- Nothing to mention.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @topeogunleye in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/App.jsx
Comment (LionRouge1) [inline src/App.jsx:1]: - [ ] According to the project requirement the `App` component should have a route that leads to the `Greeting` component. Therefore I suggest that you make use of **`react-route`** to create a route in the `App` component to lead to the Greeting component to follow the requirement
Screenshot from 2024-02-02 22-49-31

File: src/App.jsx [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+  return (
+    <div>
+      <h1>Greetings App</h1>
+      <ul>
+        {greetings.map(greeting => (
+          <li key={greeting.id}>{greeting.greeting}</li>
+        ))}
+      </ul>
+    </div>
```
Comment (LionRouge1) [inline src/App.jsx]: - [ ] Kindly use redux to get the message from the API endpoint as requested in the project requirement.
Screenshot from 2024-02-03 00-33-48

File: src/Greeting.jsx [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+// function Greeting({ username }) {
+//   return (
+//     <div>
+//       <h1>Hello, {username}!</h1>
+//     </div>
+//   );
+// }
+
+// export default Greeting;
```
Comment (LionRouge1) [inline src/Greeting.jsx]: [OPTIONAL]
- [ ] Please remove all commented out code from your code base. It make your project look messy and unclean

**Review by LionRouge1 — CHANGES_REQUESTED** [review]
Hi @gilberthappi, 

Great work on making the changes 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

almost

## Suggested changes :recycle: 

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by topeogunleye — APPROVED** [review]
### Hi @gilberthappi, 

# Status: Approved ✔️
Excellent job on making the changes requested by the previous code reviewer. Keep it up 👏

Your project is complete! There is nothing else to say other than... it's time to merge it 💪
Congratulations! 🎉
congratulations
>
## To highlight :champagne: 
- Made all changes requested :heavy_check_mark:
 
## [Optional] suggestions:
* [ ]  Please check my inline comments.
 
 _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
 
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @topeogunleye in your question so I can receive the notification.



------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**